### PR TITLE
github enterprise support for pullrequest pipeline resource

### PR DESCRIPTION
**Pull request by akihikokuroda**
The pull request resource supports only public github.  This PR adds the support for the enterprise github.  When the url in the pull request resource has the base url other than "github.com", it sets the url as the base url for the pull request.

Test for the NewGitHubHandler is added in cmd/pullrequest-init/github_test.go 

# Changes

When the url in the pull request resource has the base url other than "github.com", it sets the url as the base url for the pull request.

Code changes are in NewGitHubHandler in cmd/pullrequest-init/github.go
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
github enterprise support
```

### CODE REVIEW

File: cmd/pullrequest-init/github_test.go [hunk 1]
```diff
@@ -56,13 +60,31 @@ func TestGitHubParseURL_errors(t *testing.T) {
...
-			if o, r, pr, err := parseGitHubURL(url); err == nil {
-				t.Errorf("Expected error, got (%s, %s, %d)", o, r, pr)
+			if o, r, h, pr, err := parseGitHubURL(url); err == nil {
+				t.Errorf("Expected error, got (%s, %s, %s, %d)", o, r, h, pr)
 			}
 		})
 	}
 }
 
+func TestNewGitHubHandler(t *testing.T) {
```
Comment (wlynch) [inline cmd/pullrequest-init/github_test.go]: The 2 test cases are pretty much identical. Group these into a table driven test: https://github.com/golang/go/wiki/TableDrivenTests

File: cmd/pullrequest-init/github_test.go [hunk 2]
```diff
@@ -20,9 +20,10 @@ import (
 func TestGitHubParseURL(t *testing.T) {
 	wantOwner := "owner"
 	wantRepo := "repo"
+	wantHost := []string{"https://github.com", "https://github.com", "https://github.com", "http://github.com", "ssh://github.com", "https://example.com", "https://github.com",}
```
Comment (wlynch) [inline cmd/pullrequest-init/github_test.go]: Keeping 2 separate arrays makes it unclear where how belongs to which URL. Instead, change the `url := range []string` below into:

```
tests := []struct {
  url string
  wantHost string
} {
 ...
}
for _, tc := range tests {
  ...
} 
```

See https://github.com/golang/go/wiki/TableDrivenTests for more details.

File: cmd/pullrequest-init/github_test.go [hunk 3]
```diff
@@ -279,14 +301,15 @@ func TestParseGitHubURL(t *testing.T) {
 		fmt.Sprintf("https://github.com/%s/%s/pull/%d/foo", owner, repo, prNum),
 		fmt.Sprintf("github.com/%s/%s/pull/%d/foo", owner, repo, prNum),
 	}
-	for _, u := range good {
+	host := []string{"https://github.com", "https://github.com", "http://github.com", "tacocat://github.com", "https://example.com", "https://github.com", "://",}
```
Comment (wlynch) [inline cmd/pullrequest-init/github_test.go]: Also change this to a table driven test.

File: cmd/pullrequest-init/github.go [hunk 4]
```diff
@@ -40,13 +40,21 @@ func NewGitHubHandler(ctx context.Context, logger *zap.SugaredLogger, rawURL str
...
 
-	owner, repo, prNumber, err := parseGitHubURL(rawURL)
+	owner, repo, host, prNumber, err := parseGitHubURL(rawURL)
 	if err != nil {
 		return nil, err
 	}
-
+	var client *github.Client
+	if !strings.Contains(host, "github.com") {
+		client, err = github.NewEnterpriseClient(host + "/api/v3/", host + "/api/v3/", hc)
```
[the remark this answers is not in the document]
Comment (akihikokuroda) [reply cmd/pullrequest-init/github.go]: I update.

File: cmd/pullrequest-init/github.go [hunk 5]
```diff
@@ -40,13 +40,21 @@ func NewGitHubHandler(ctx context.Context, logger *zap.SugaredLogger, rawURL str
 		hc = oauth2.NewClient(ctx, ts)
 	}
 
-	owner, repo, prNumber, err := parseGitHubURL(rawURL)
+	owner, repo, host, prNumber, err := parseGitHubURL(rawURL)
```
[the remark this answers is not in the document]
Comment (akihikokuroda) [reply cmd/pullrequest-init/github.go]: I understood.  I'll think about updating it later.

File: cmd/pullrequest-init/github_test.go [hunk 6]
```diff
@@ -56,13 +60,31 @@ func TestGitHubParseURL_errors(t *testing.T) {
...
+		t.Fatalf("error creating GitHubHandler: %v", err)
+	}
+	if h.BaseURL.Host != "api.github.com" {
+		t.Fatalf("error unexpected BaseURL.Host: %v", h.BaseURL.Host)
+	}
+	h, err = NewGitHubHandler(ctx, zaptest.NewLogger(t, zaptest.WrapOptions(zap.AddCaller())).Sugar(), "https://github.tekton.com/tektoncd/pipeline/pull/1")
+	if err != nil {
+		t.Fatalf("error creating GitHubHandler: %v", err)
+	}
+	if h.BaseURL.Host != "github.tekton.com" {
```
[the remark this answers is not in the document]
Comment (akihikokuroda) [reply cmd/pullrequest-init/github_test.go]: I update it.

File: cmd/pullrequest-init/github_test.go (hunk 1 above)
Comment (akihikokuroda) [reply cmd/pullrequest-init/github_test.go]: I update it.

File: cmd/pullrequest-init/github_test.go (hunk 2 above)
Comment (akihikokuroda) [reply cmd/pullrequest-init/github_test.go]: Yes, it's better.  I update it.

File: cmd/pullrequest-init/github_test.go (hunk 3 above)
Comment (akihikokuroda) [reply cmd/pullrequest-init/github_test.go]: I update it.

File: cmd/pullrequest-init/github.go [hunk 7]
```diff
@@ -57,22 +65,22 @@ func NewGitHubHandler(ctx context.Context, logger *zap.SugaredLogger, rawURL str
 // parseURL takes in a raw GitHub URL
 // (e.g. https://github.com/owner/repo/pull/1) and extracts the owner, repo,
```
[the remark this answers is not in the document]
Comment (akihikokuroda) [reply cmd/pullrequest-init/github.go]: I update it.

**Comment by vdemeester** [comment]
Fixing `I0703 07:49:20.088] ERROR: test demo-pipeline-run-1-deploy-web-rdhvx=SucceededFalse but should be succeededtrue` in https://github.com/tektoncd/pipeline/pull/1038

**Pull request merged**